ambry-store/src/test/java/com.github.ambry.store/IndexSegmentTest.java

```diff
@@ -231,8 +292,13 @@ private void addPutEntries(List<Long> offsets, long lastEntrySize, IndexSegment
       long offset = offsets.get(i);
       long size = i == offsets.size() - 1 ? lastEntrySize : offsets.get(i + 1) - offset;
       IndexValue value =
-          new IndexValue(size, new Offset(segment.getLogSegmentName(), offset), (byte) 0, Utils.Infinite_Time);
+          IndexValueUtils.getIndexValue(size, new Offset(segment.getLogSegmentName(), offset), time.seconds(), version);
       segment.addEntry(new IndexEntry(id, value), new Offset(segment.getLogSegmentName(), offset + size));
+      if (resetKey == null) {
```

Comment:
It is only coincidental that it mimics the `IndexSegment`. However the specific concern I have with this is that modelling something as a class variable sometimes makes the test harder to understand. In this case, I wouldn't guess that `addPutEntries()` would do this. Returning all the entries added is not only useful but also lets the caller (the actual test) decide what the reset key should be without coupling that logic with the `addPutEntries()` logic.